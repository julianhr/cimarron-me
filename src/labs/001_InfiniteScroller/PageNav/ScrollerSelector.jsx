import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { setScrollerType } from '../actions/rootActions'
import Selector from './Selector'


export const SCROLLERS = [
  { intersectionObserver: 'Intersection Observer with Sentinel' },
  { containerScrollHeights: 'Container scrollHeight and scrollTop' },
  { sentinelClientRect: 'Sentinel bounding client rectangle' },
]

export function ScrollerSelector({ scrollerType, setScrollerType }) {
  return (
    <Selector
      label='Scroller Re-Fetch Method:'
      keys={SCROLLERS.map(scroller => Object.keys(scroller))}
      titles={SCROLLERS.map(scroller => Object.values(scroller))}
      selected={scrollerType}
      onChange={setScrollerType}
    />
  )
}

ScrollerSelector.propTypes = {
  setScrollerType: PropTypes.func.isRequired,
  scrollerType: PropTypes.string.isRequired,
}

const mapStateToProps = ({ scrollerType }) => ({ scrollerType })
const mapDispatchToProps = { setScrollerType }

export default connect(mapStateToProps, mapDispatchToProps)(ScrollerSelector)
