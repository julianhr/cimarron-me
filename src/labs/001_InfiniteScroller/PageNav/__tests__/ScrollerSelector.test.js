import React from 'react'
import { render, mount, shallow } from 'enzyme'
import { create } from 'react-test-renderer'

import { ScrollerSelector, SCROLLERS } from '../ScrollerSelector'
import RootMock from '../../__tests__/__mocks__/RootMock'


describe('ScrollerSelector', () => {
  const testProps = {
    setScrollerType: () => {},
    scrollerType: Object.keys(SCROLLERS[0])[0],
  }

  const renderer = (props) => mount(<RootMock><ScrollerSelector {...props} /></RootMock>)

  describe('props', () => {
    test('setScrollerType is required', () => {
      const props = {...testProps}
      delete props.setScrollerType
      expect(() => renderer(props)).toThrow()
    })

    test('scrollerType is required', () => {
      const props = {...testProps}
      delete props.scrollerType
      expect(() => renderer(props)).toThrow()
    })
  })

  describe('snapshots', () => {
    it('matches with all props', () => {
      const wrapper = renderer(testProps)
      expect(wrapper).toMatchSnapshot()
      wrapper.unmount()
    })
  })
})
