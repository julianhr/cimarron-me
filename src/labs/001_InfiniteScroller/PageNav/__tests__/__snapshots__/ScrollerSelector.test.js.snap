// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollerSelector snapshots matches with all props 1`] = `
Array [
  <ForwardRef(render)
    styles={
      Object {
        "map": undefined,
        "name": "1n3w7e7-globalStyles-globalStyles",
        "next": undefined,
        "styles": "body{color:#222222,lineHeight 1.3em;font-size:18px;}input,textarea{border:2px solid #BBB;background:#f7f7f7;border-radius:5px;font-size:20px;}input,textarea{border:1px solid #BBB;background:#f7f7f7;font-size:1.1em;padding:4px 8px;}p,h1,h2,h3,h4,h5,h6,figure{margin:0;}h1,h2,h3,h4,h5,h6{color:#9B1518;}p{padding:.8em 0;line-height:1.4em;}h1{font-size:36px;margin:50px 0 40px;}h2{line-height:1.4em;font-size:30px;font-weight:600;margin:40px 0 30px;}h3{font-size:24px;font-weight:600;margin:30px 0 30px;}h4{line-height:1.3em;font-size:18px;}h5{font-size:12px;}h6{font-size:8px;}code{color:#D90000;font-size:14;}label:globalStyles;label:globalStyles;",
      }
    }
  />,
  <Root>
    <div
      className="css-1o7hb61-Root el5yv3l0"
    >
      <label
        htmlFor="scroller-type"
      >
        Scroller Re-Fetch Method:
      </label>
      <select
        id="scroller-type"
        onChange={[Function]}
        value="intersectionObserver"
      >
        <option
          key="intersectionObserver"
          value="intersectionObserver"
        >
          Intersection Observer with Sentinel
        </option>
        <option
          key="containerScrollHeights"
          value="containerScrollHeights"
        >
          Container scrollHeight and scrollTop
        </option>
        <option
          key="sentinelClientRect"
          value="sentinelClientRect"
        >
          Sentinel bounding client rectangle
        </option>
      </select>
    </div>
  </Root>,
]
`;
